Only admins can remove users from an event

ThuTrinh opened the pull request:
## Description
When displaying the list of users that have signed up for an event, only show the remove button if the current signed in user is an admin.

## References
[Github Issue](https://github.com/zendesk/volunteer_portal/issues/311)

## Screenshots
![vp-admin-only-delete-after](https://user-images.githubusercontent.com/3032977/73836346-26e93000-4863-11ea-9bbe-f8e7bf816617.gif)

## Implementation Details
The fix for this issue is only relevant in 1 file: `app/javascript/pages/Event/index.js`

I decided to just refactor the components a bit to make it clearer and more maintainable.

## CCs
@zendesk/volunteer

## Risks (if any)
* [low] - minor change to hide a button, should not effect anything else.

ThuTrinh committed:
:bug: Only admins can remove users from an event

When displaying the list of users that have signed up for an event, only show the remove button if the current signed in user is an admin.

kennethc commented on app/javascript/components/UserList/index.js:
```diff
@@ -45,14 +47,16 @@ const UserList = ({ users, destroySignup }) => {
-                </Icon>
-              </IconButton>
+              {currentUser.isAdmin && (
+                <IconButton
```
This is technically correct, but is it possible to use an `if` block here? I think it will be clearer to a wider audience what we are trying to achieve here.

ThuTrinh replied:
sure :)

ThuTrinh replied:
I gave this more thought and while I do agree with you, I think that we should keep the statement as is for the following reasons:
1. This technique of rendering logic is idiomatic in react and will be present in many codebases. We should expose people to it and help them use it appropriately rather than avoid it imo.
2. The react [documentation](from './UserList') encourages it anyway, with a caveat that the value before the `&&` must be a boolean. We should look for this in code reviews.
3. The alternative `if` statement actually makes the code look more complicated imo. 
e.g.
```javascript
// inline
{currentUser.isAdmin ? (
    <IconButton aria-label={`Remove volunteer ${user.name} from this event`} onClick={confirm}>
        <Icon>
          <RemoveIcon />
        </Icon>
      </IconButton>
    ) : null 
}
```

ThuTrinh committed:
:recycle: abstract out remove btn and modal as their own components

The motivation here is to clearly separate the responsibilies for better composability and maintenance

Pancrisp reviewed (approved):
Agree with @ThuTrinh's statement above. The logical `&&` operator is a standard approach to conditional rendering in React and is preferred over the ternary operator if your false conditional doesn't output anything.
